hazelcast/src/main/java/com/hazelcast/spi/impl/proxyservice/impl/ProxyRegistry.java

```diff
@@ -262,11 +263,26 @@ void destroy() {
             if (!future.isSetAndInitialized()) {
                 continue;
             }
-            DistributedObject distributedObject = future.get();
-            if (distributedObject instanceof AbstractDistributedObject) {
-                ((AbstractDistributedObject) distributedObject).invalidate();
-            }
+
+            DistributedObject distributedObject = extractDistributedObject(future);
+            invalidate(distributedObject);
         }
         proxies.clear();
     }
+
+    private DistributedObject extractDistributedObject(DistributedObjectFuture future) {
+        try {
+            return future.get();
+        } catch (Throwable ex) {
+            EmptyStatement.ignore(ex);
```

Comment:
won't this eat the Interrupted flag when InterruptedException is thrown?

Reply:
Good question. It won't because this future [doesn't throw InterruptedException](https://github.com/hazelcast/hazelcast/blob/master/hazelcast/src/main/java/com/hazelcast/spi/impl/proxyservice/impl/DistributedObjectFuture.java#L35) and the interrupted flag is not cleared. 

Another question though is what happens if that `future.get()` meets with an interrupt without cached result or error. I'm taking a look.

Reply:
`future.get()` doesn't handle interrupts correctly, created #12585 for this.